ml/Host.cc

```diff
@@ -440,6 +490,41 @@ void DetectableHost::detect() {
     }
 }
 
+void DetectableHost::getAnomalyRateInfoCurrentRange(std::vector<std::pair<std::string, double>> &V, time_t After, time_t Before) {
+    {
+        std::lock_guard<std::mutex> Lock(Mutex);
+        for (auto &DP : DimensionsMap) {
+            Dimension *D = DP.second;
+            V.push_back({D->getID(), (D->getAnomalyPercentage() * abs(Before - After) / (SaveAnomalyPercentageEvery * static_cast<double>(updateEvery())))});
+        }
+    }
+}
+
+void DetectableHost::getAnomalyRateInfoMixedRange(std::vector<std::pair<std::string, double>> &V, std::string HostUUID,time_t After, time_t Before) {
+    std::vector<std::pair<std::string, double>> DimAndAnomalyRateInRange;
+    bool Res = getAnomalyRateInfoInRange(DimAndAnomalyRateInRange, HostUUID, After, getLastSavedBefore());
+    std::vector<std::pair<std::string, double>>::iterator it;
+    
+    if (Res) {
+        {
+        std::lock_guard<std::mutex> Lock(Mutex);
+            for (auto &DP : DimensionsMap) {
+                Dimension *Dm = DP.second;
```

Comment:
Any reason to name the variable `Dm` and not `D` like we do in the rest of the code?

Reply:
was in test mode! changed.